x-pack/plugin/ml-package-loader/src/main/java/org/elasticsearch/xpack/ml/packageloader/action/TransportLoadTrainedModelPackage.java

```diff
@@ -91,6 +94,17 @@ protected ClusterBlockException checkBlock(Request request, ClusterState state)
     @Override
     protected void masterOperation(Task task, Request request, ClusterState state, ActionListener<AcknowledgedResponse> listener)
         throws Exception {
+        if (existingDownloadInProgress(request.getModelId(), request.isWaitForCompletion(), listener)) {
+            logger.debug("Existing download of model [{}] in progress", request.getModelId());
+
+            if (request.isWaitForCompletion() == false) {
+                listener.onResponse(AcknowledgedResponse.TRUE);
```

Comment:
I find the "ownership" of the `listener` a bit confusing. Depending on the result and "wait for completion", either the `existingDownloadInProgress` method take cares of the listener or doesn't.

I think it's more clear if this is just:
```
if (existingDownloadInProgress(...)) {
    return;
}
```
and move the `onResponse` to `existingDownloadInProgress`.

Reply:
And maybe rename to `handleDownloadInProgress` or so to make it more clear that this does stuff, and not just return true/false.